tangos/tools/consistent_trees_importer.py

```diff
@@ -34,12 +35,32 @@ def process_options(self, options):
 
 
     @staticmethod
-    def filename_to_snapnum(filename):
+    def filename_to_snapnum(filename,basedir):
         match = re.match(".*(?:snapdir|snapshot)_([0-9]+)/?$", filename)
         if match:
             return int(match.group(1))
         else:
-            raise ValueError("Unable to convert %s to snapshot number"%filename)
+            match = re.match("(?:RD|DD)[0-9]*/(?:RD|DD)([0-9]*)",filename)
+            if match:
+                # Check whether datasets.txt exists (i.e., if rockstar was run with yt)
+                if os.path.exists(os.path.join(basedir, "datasets.txt")):
+                    with open(os.path.join(basedir, "datasets.txt")) as f:
```

Comment:
I notice you've used this `datasets.txt` files in a few different places. It may be that the logic about its use deserves to be in a utility function somewhere, and called from all the different places rather than reimplemented each time. Not 100% necessary, just would be neater if it's possible.

Reply:
I took a shot at writing a quick utility function for this. For now, it's in tangos/util/read_datasets_file.py. Is this the right place for it? (and does the function itself look okay? It seems to work as expected, but I know there might be a more efficient way to accomplish the same thing)